src/wagtail_2fa/forms.py

```diff
@@ -68,7 +69,9 @@ def clean_otp_token(self):
 
     def clean_password(self):
         password = self.cleaned_data.get("password")
-        if not self.user.check_password(password):
+        if not authenticate(
+            self.request, username=self.request.user.username, password=password
```

Comment:
I tried adding a new device to a user but it failed, raising an `AttributeError` on `self.request.user.username`. `self.request.user` returns a `SimpleLazyObject` representing the `User` model, which does not have that attribute.

After looking at the docs, I found [this](https://docs.djangoproject.com/en/dev/ref/contrib/auth/#django.contrib.auth.models.User.get_username), which says the `get_username()` method should be used instead of referencing the `username` attribute directly.

When I replace `username=self.request.user.username` with `username=self.request.user.get_username()`, it works fine on my end. However I am not sure if this works with your custom authentication implementation. Could you check that for me? If that works correctly, this PR is good to go.

Reply:
I'm not sure how to test this locally - any advice here @nnist?

Reply:
I suppose the fastest way to test it would be to temporarily change that line in the wagtail-2fa source code in your local virtualenv for the project you're using this package in.

This change should technically not break anything though - all User models are supposed to have that `get_username()` method, and the docs recommend using it. It works fine in the sandbox. We could simply commit this change and merge it and you could test it later in the next release, if that's easier for you.

Reply:
That's easier for me - let's do that @nnist